lsp/src/uri0.ml

```diff
@@ -1,31 +1,112 @@
+(* This module is based on the [vscode-uri] implementation:
+   https://github.com/microsoft/vscode-uri/blob/main/src/uri.ts. It only
+   supports scheme, authority and path. Query, port and fragment are not
+   implemented *)
+
 open Import
 
 module Private = struct
   let win32 = ref Sys.win32
 end
 
-type t = Uri_lexer.t =
```

Comment:
if you keep this type equality, you will not need to do the conversion from `Uri_lexer.t` to `Uri.t`

Reply:
Nice! 👌